src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -564,26 +564,45 @@ void VM_Version::initialize() {
   check_virtualizations();
 }
 
-void VM_Version::check_virtualizations() {
 #if defined(LINUX)
-  const char* info_file = "/sys/devices/virtual/dmi/id/product_name";
-  // check for various strings in the dmi data indicating virtualizations
+static int check_info_file(const char* fpath, const char* virt1,
+                           const char* virt2) {
   char line[500];
-  FILE* fp = os::fopen(info_file, "r");
+  FILE* fp = os::fopen(fpath, "r");
   if (fp == nullptr) {
-    return;
+    return false;
   }
   while (fgets(line, sizeof(line), fp) != nullptr) {
-    if (strcasestr(line, "KVM") != 0) {
-      Abstract_VM_Version::_detected_virtualization = KVM;
-      break;
+    if (strcasestr(line, virt1) != 0) {
+      fclose(fp);
+      return 1;
     }
-    if (strcasestr(line, "VMware") != 0) {
-      Abstract_VM_Version::_detected_virtualization = VMWare;
-      break;
+    if (virt2 != NULL && strcasestr(line, virt2) != 0) {
+      fclose(fp);
+      return 2;
     }
   }
   fclose(fp);
+  return 0;
+}
+#endif
+
+void VM_Version::check_virtualizations() {
+#if defined(LINUX)
+  const char* pname_file = "/sys/devices/virtual/dmi/id/product_name";
+  const char* tname_file = "/sys/hypervisor/type";
+  int resp = check_info_file(pname_file, "KVM", "VMWare");
+  if (resp == 1) {
+    Abstract_VM_Version::_detected_virtualization = KVM;
```

Comment:
This may be what Christoph suggested by why not keep this in check_info_file so you don't need to map return codes. check_info_file could just be a bool function then.